File: backend/models/companyModel.js
const mongoose = require('mongoose');

const company = new mongoose.Schema({
    fname : {
        type : String,
        require : true,
    },
    lname : {
        type : String,
        require : true
    },
    username : {
        type : String,
        require : true
    },
    email: {
        type : String,
        require : true
    },
    phone : {
        type : Number,
        require : true
    },
    address : {
        type : String
    },
    password : {
        type : String,
        require : true
    },
    picture : {
        type : String,
    }
})

module.exports = mongoose.model("Company", company)

File: frontend/lib/models/companyModel.dart
import 'package:json_annotation/json_annotation.dart';

class CompanyModel {
  @JsonKey(name: "_id")
  String? id;
  String? fname;
  String? lname;
  String? gender;
  int? age;
  String? username;
  String? email;
  int? phone;
  String? address;
  String? password;
  String? department;
  String? picture;

  CompanyModel({
    this.id,
    this.fname,
    this.lname,
    this.gender,
    this.age,
    this.username,
    this.email,
    this.phone,
    this.address,
    this.password,
    this.department,
    this.picture,
  });

  // converting to dart object from server json file
  factory CompanyModel.fromJson(Map<String, dynamic> json) => CompanyModel(
        id: json["_id"],
        fname: json["fname"],
        lname: json["lname"],
        gender: json["gender"],
        age: json["age"],
        username: json["username"],
        email: json["email"],
        phone: json["phone"],
        address: json["address"],
        password: json["password"],
        department: json["department"],
        picture: json["picture"],
      );

  // converting to json format from dart object
  Map<String, dynamic> toJson() => {
        "_id": id,
        "fname": fname,
        "lname": lname,
        "gender": gender,
        "age": age,
        "username": username,
        "email": email,
        "phone": phone,
        "address": address,
        "password": password,
        "department": department,
        "picture": picture,
      };
}
